### DISCUSSION TITLE
sqltargets - shorthand for SQL queries [show-and-tell]

### DISCUSSION THREAD

**Discussion by daranzolin** [ropensci/targets#1265] [Show and tell]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Hi everyone, I fall in love with targets more and more every day, but have always thought running independent SQL files in a targets project was never as smooth as I'd like. Towards that end, I just created [sqltargets,](https://github.com/daranzolin/sqltargets) a targets extension that offers a `tar_sql` shorthand. The majority of the code has been adapted from `tar_quarto`, but some of it remains beyond my comprehension. In any case, I would love some critiques and feedback from you experts! The package has been minimally tested against some local SQLite databases.

**Comment by wlandau** [comment]
Great package! `tar_sql()` looks like a convenient abstraction, and I like how you structured the internals. My only suggestion would be to consider exposing the `format` argument of `tar_target_raw()` (currently supplied by `tar_option_set()`: https://github.com/daranzolin/sqltargets/blob/d20f8b65359701293f5e65d24d83e58029cbba1a/R/tar_sql_raw.R#L74).

**Reply by daranzolin** [reply]
Thanks so much Will, I think the format argument is creating some difficulty for me. `tar_make()` currently does not rerun the query when the query file changes, and my understanding is that this is because the format does not equal 'file'. I still want, however, the returned value to be the compressed data frame (e.g. rds, qs, etc.) that is the result of the query. Is there a way I can accomplish both? That is, given `tar_sql(report, "query.sql")`, `tar_read(report)` is a data frame, but targets will rerun the query when "query.sql" changes? Perhaps I am trying to have my cake and eat it too.

**Reply by wlandau** [reply]
You could have `tar_sql()` create two different targets: an upstream one which tracks the file, and a downstream one which uses it and runs the query.

**Reply by daranzolin** [reply]
That works great, thanks! sqltargets is now on CRAN, and I did take the liberty of adding the 'Targetopia' badge...hope that wasn't presumptuous of me.

**Reply by wlandau** [reply]
Fantastic! Feel free to also add your package to https://wlandau.github.io/targetopia/packages.html. The source file of that page is https://github.com/wlandau/targetopia/blob/main/packages.Rmd.
